src/wallet_client.ts

```diff
@@ -174,32 +174,56 @@ export class WalletClient {
   async transferSuiMnemonic(
     amount: number,
     suiAccount: Ed25519Keypair,
-    receiverAddress: SuiAddress
+    receiverAddress: SuiAddress,
+    typeArg: string = SUI_TYPE_ARG
   ) {
     const keypair = suiAccount;
     const senderAddress = keypair.getPublicKey().toSuiAddress();
-    const coinsNeeded =
+    if(typeArg === SUI_TYPE_ARG){
+      const coinsNeeded =
       await this.provider.selectCoinSetWithCombinedBalanceGreaterThanOrEqual(
         senderAddress,
-        BigInt(amount + DEFAULT_GAS_BUDGET_FOR_SUI_TRANSFER)
+        BigInt(amount + DEFAULT_GAS_BUDGET_FOR_SUI_TRANSFER),
+        typeArg
       );
-    const inputCoins: ObjectId[] = coinsNeeded.map((coin) => getObjectId(coin));
-    const recipients: SuiAddress[] = [receiverAddress];
-    const amounts: number[] = [amount];
-    const payTxn: PaySuiTransaction = {
-      inputCoins: inputCoins,
-      recipients: recipients,
-      amounts: amounts,
-      gasBudget: DEFAULT_GAS_BUDGET_FOR_SUI_TRANSFER,
-    };
-    const signer = new RawSigner(keypair, this.provider, this.serializer);
-    return await signer.paySui(payTxn);
+      const inputCoins: ObjectId[] = coinsNeeded.map((coin) => getObjectId(coin));
+      const recipients: SuiAddress[] = [receiverAddress];
+      const amounts: number[] = [amount];
+      const payTxn: PayTransaction = {
+        inputCoins: inputCoins,
+        recipients: recipients,
+        amounts: amounts,
+        gasBudget: DEFAULT_GAS_BUDGET_FOR_SUI_TRANSFER,
+      };
+      const signer = new RawSigner(keypair, this.provider, this.serializer);
+      return await signer.pay(payTxn);
+    }else{
+      const coinsNeeded =
+      await this.provider.selectCoinSetWithCombinedBalanceGreaterThanOrEqual(
+        senderAddress,
+        BigInt(amount),
+        typeArg
+      );
+      const inputCoins: ObjectId[] = coinsNeeded.map((coin) => getObjectId(coin));
+      const gasObjId = await this.getGasObject(senderAddress, inputCoins);
+      const recipients: SuiAddress[] = [receiverAddress];
+      const amounts: number[] = [amount];
+      const payTxn: PayTransaction = {
+        inputCoins: inputCoins,
+        recipients: recipients,
+        amounts: amounts,
+        gasPayment: gasObjId,
+        gasBudget: DEFAULT_GAS_BUDGET_FOR_SUI_TRANSFER,
```

Comment:
do we need both gasPayment and gasBudget?

Reply:
yes gas payment refers the gas object and gas budget amount is used from that object